convert.py

```diff
@@ -1116,8 +1184,17 @@ def main(args_in: Optional[List[str]] = None) -> None:
     if args.dump_single:
         model_plus = lazy_load_file(args.model)
         do_dump_model(model_plus)
+        return
 
-    model_plus = load_some_model(args.model)
+    if not args.vocab_only:
```

Comment:
There are some changes to the behavior in this area. I don't know if it was intended you could do stuff like `--dump-single` _and_ `--vocab-only` to do both those things at the same time. It seems weird and unlikely and handling the different possibilities complicated the code.  The options are basically exclusive now.